vassal-app/src/main/java/VASSAL/configure/ConfigureTree.java

```diff
@@ -1487,21 +1792,302 @@ private int getNumMatches(String searchString) {
+          chat("&nbsp;&nbsp;{" + idString + "} " + ((desc != null) ? desc : ""));
+        }
+      }
+    }
+
+
+    private void showConfigurableHitList(DefaultMutableTreeNode node, String searchString) {
+      final Configurable c = (Configurable) node.getUserObject();
+      String name = (c.getConfigureName() != null ? c.getConfigureName() : "") +
+        " [" + getConfigureName(c.getClass()) + "]";
+
+      TargetProgress progress = new TargetProgress();
+      String matchString = "<b><u>Matches for " + name + ": </u></b>";
+
+      if (!(c instanceof SearchTarget)) {
+        return;
+      }
+
+      SearchTarget st = (SearchTarget) c;
+      String desc = getConfigureName(c.getClass());
+
+      if (searchParameters.isMatchExpressions()) {
+        List<String> exps = st.getExpressionList();
+        if (exps != null) {
+          for (String s : exps) {
+            if (!StringUtils.isEmpty(s) && checkString(s, searchString)) {
+              progress.checkShowTrait(matchString, desc, "");
+              chat("&nbsp;&nbsp;&nbsp;&nbsp;{Expression} " + s);
+            }
+          }
+        }
+      }
+
+      if (searchParameters.isMatchProperties()) {
+        List<String> props = st.getPropertyList();
+        if (props != null) {
+          for (String s : props) {
+            if (!StringUtils.isEmpty(s) && checkString(s, searchString)) {
+              progress.checkShowTrait(matchString, desc, "");
+              chat("&nbsp;&nbsp;&nbsp;&nbsp;{Property} " + s);
+            }
+          }
+        }
+      }
+
+      if (searchParameters.isMatchKeys()) {
+        List<NamedKeyStroke> keys = st.getNamedKeyStrokeList();
+        if (keys != null) {
+          for (NamedKeyStroke k : keys) {
+            if (k != null) {
+              String s = k.isNamed() ? k.getName() : KeyNamer.getKeyString(k.getStroke());
+              if (!StringUtils.isEmpty(s) && checkString(s, searchString)) {
+                progress.checkShowTrait(matchString, desc, "");
+                chat("&nbsp;&nbsp;&nbsp;&nbsp;{KeyCommand} " + s);
+              }
+            }
+          }
+        }
+      }
+
+      if (searchParameters.isMatchMenus()) {
+        List<String> menus = st.getMenuTextList();
+        if (menus != null) {
+          for (String s : menus) {
+            if (!StringUtils.isEmpty(s) && checkString(s, searchString)) {
+              progress.checkShowTrait(matchString, desc, "");
+              chat("&nbsp;&nbsp;&nbsp;&nbsp;{UI Text} " + s);
+            }
+          }
+        }
+      }
+
+      if (searchParameters.isMatchMessages()) {
+        List<String> msgs = st.getFormattedStringList();
+        if (msgs != null) {
+          for (String s : msgs) {
+            if (!StringUtils.isEmpty(s) && checkString(s, searchString)) {
+              progress.checkShowTrait(matchString, desc, "");
+              chat("&nbsp;&nbsp;&nbsp;&nbsp;{Message/Field} " + s);
+            }
+          }
+        }
+      }
+    }
+
```

Comment:
These are rather repetitive and cry out for a unifying method.

Reply:
Okay, yes, those look an ass-ton better now.